src/duplicacy_utils.go

```diff
@@ -56,10 +56,21 @@ func IsUnspecifiedFilter(pattern string) bool {
 }
 
 func IsValidRegex(pattern string) (valid bool, err error) {
-    _, err = regexp.Compile(pattern)
+
+    var re *regexp.Regexp = nil
+
+    if re, valid = RegularExpressions[pattern]; valid && re != nil {
+        LOG_WARN("REGEX_FOUND", "Found already compiled regex for pattern \"%s\", regexp=%#v", pattern, re)
```

Comment:
I understand the logic behind why you added this line, but it makes it such that this method can only be meaningfully used in the context of a single pass through a single group of regexes that are supposed to have only one instance of each one.

A better approach would be to test the raw strings for duplicates, leaving duplicate-detection in the hands of the code that has the full list and keeping this method focused solely on determining if the regex is valid (and populating the cache if it's a new regex).

The duplicate-detection in the calling method could be as simple as putting all values into a set and checking whether it has the same size as the original collection; fewer entries in the set means that there is at least one set of duplicates in the batch.

Reply:
The message is simply meant to convey that we've already encountered this same regex pattern before, which may (probably) indicates that the user has duplicate filters specified. I made it a warning instead of error, because after the first one is matched, no further filters are matched, so it shouldn't affect the overall logic, etc.

Mostly it was just a sanity check I put in to verify what I was doing with the compiled regex's and not re-compiling them over and over, etc.

Reply:
While you were writing the code, I can see using WARN as the logging level. But now that you've got it working, it should drop to DEBUG.

The method simply determines if the regex is valid, caching it if it's not already cached. Nothing about that logic says that seeing the same regex more than once (over the entire lifetime of the program) is invalid or worthy of a WARN/ERROR. Only the code that's assembling the list of regexes to be evaluated cares whether you've seen this regex before, so if anyone's going to be throwing WARNs/ERRORs, it's that method.

So if you care about checking for (and logging) that condition, add it to that method. If not, then don't worry about it. But either way, make this line a DEBUG, because this method shouldn't care about the business logic of that other method.

Reply:
I think I'd prefer to just remove the message altogether.

Reply:
I'm good with that.

Reply:
it's gone, i think that addresses all of your comments.

Reply:
The only remaining comment I had was that I agreed with your suggestion to move the map from duplicacy_regex.go to duplicacy_utils.go and get rid of duplicacy_regex.go.